### PR TITLE
init nixos-remote

### PR DESCRIPTION
**Pull request by Lassulus**
initial version of nixos-remote. takes a flake like https://github.com/Lassulus/flakes-testing to format/install

### CODE REVIEW

File: nixos-remote [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    curl --fail -Ss -L "\$1"
+  elif command -v wget >/dev/null 2>&1; then
+    wget "\$1" -O-
+  else
+    echo "no downloader (curl or wget) found, bailing out"
+    exit 1
+  fi
+}
+rm -rf /root/kexec
+mkdir -p /root/kexec
```
Comment (Mic92) [inline nixos-remote:88]: Maybe useful?

```suggestion
if command -v tar >/dev/null 2>&1; then
    echo "no tar command found, but required to unpack kexec tarball" >&2
    exit 1
fi
```

**Comment by Lassulus** [comment]
possible next steps:
- non flake support (do we want this?)
- absorb https://github.com/numtide/terraform-nixos-install
- absorb https://github.com/nix-community/nixos-images
- documentation

**Comment by Mic92** [comment]
I would say README is the most important one. Than I can have people starting to test this.

File: nixos-remote (hunk 1 above)
Comment (Lassulus) [reply nixos-remote:95]: alright, was absorbed

File: nixos-remote [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    wget "\$1" -O-
+  else
+    echo "no downloader (curl or wget) found, bailing out"
+    exit 1
+  fi
+}
+if command -v tar >/dev/null 2>&1; then
+  echo "no tar command found, but required to unpack kexec tarball" >&2
+  exit 1
+fi
```
Comment (Mic92) [inline nixos-remote:93]: os=$(uname)
if [[ "$os" != "Linux" ]]; then
  echo "Expected Linux operating system, but got a '${os}'" >&2
  exit 1
fi

Comment (Lassulus) [reply nixos-remote:93]: do we expect linux? does kexec not work on other OSes? :D

File: nixos-remote [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+USAGE
+}
+
+abort() {
+  echo "aborted: $*" >&2
+  exit 1
+}
+
+nix_args=()
+kexec_url=https://github.com/nix-community/nixos-images/releases/download/nixos-22.05/nixos-kexec-installer-x86_64-linux.tar.gz
```
[the remark this answers is not in the document]
Comment (Lassulus) [reply nixos-remote:28]: maybe we just make it dependant on uname -m from the target?

[the remark this answers is not in the document]
Comment (Mic92) [reply nixos-remote:28]: target="$(uname -s).$(uname -m)"
case "$target" in
    Linux.x86_64)
        url="https://github.com/nix-community/nixos-images/releases/download/nixos-22.05/nixos-kexec-installer-x86_64-linux.tar.gz"
        ;;
     *)
       echo "$target is currently not supported by by our kexec tarball"
       exit 1
esac

**Pull request merged**